code/projects/app/src/app/stages/e5/e5-services/png-meta.service.ts

```diff
@@ -0,0 +1,97 @@
+import { Injectable } from "@angular/core";
+import { Buffer } from "buffer";
+import * as crc32 from "crc-32";
+
+@Injectable({
+  providedIn: "root",
+})
+export class PngMetaService {
+  private readonly KEY_PREFIX = "SecretText";
+  private readonly CHUNK_TYPE_VALUE = "tEXt";
+  private readonly CHUNK_TYPE = Buffer.from(this.CHUNK_TYPE_VALUE, "ascii");
+  private readonly CHUNK_TYPE_LENGTH = 4;
+  private readonly CHUNK_LENGTH_LENGTH = 4;
+  private readonly CHUNK_CRC_LENGTH = 4;
+  private readonly IEND_LENGTH = 12;
+  private readonly PNG_HEADER_LENGTH = 8;
+
+  private createKey = (key: string) => `${this.KEY_PREFIX}${key}`;
+
+  constructor() {}
+
+  public async encode(
+    inputPng: File,
+    key: string,
+    text: string,
+  ): Promise<File> {
+    const keyWithPrefix = this.createKey(key);
+    const inputBuffer = await inputPng.arrayBuffer().then((arrayBuffer) => {
+      return Buffer.from(arrayBuffer);
+    });
```

Comment:
Why not just this?

```suggestion
    const arrayBuffer = await inputPng.arrayBuffer();
    const inputBuffer = Buffer.from(arrayBuffer);
```

or even

```suggestion
    const inputBuffer = Buffer.from(await inputPng.arrayBuffer());
```

Reply:
ok

Reply:
![image](https://github.com/user-attachments/assets/3d687fea-5235-4a4a-9c8d-1ef7753d64fd)